### PR TITLE
Add an argument parser

### PR DESCRIPTION
**Pull request by tobyhodges**
This adds an `argparse.ArgumentParser` object to handle command line options gracefully. It will provide a help message, which would partially address #18 

I have also added a function to guess the output file format based on the file extension

- [x] Implement `argparse.ArgumentParser` for handling arguments+options from the command line
- [x] Guess format of output file
- [x] Update tests to reflect changes to command line structure

### CODE REVIEW

File: src/cff2pages/cff2pages.py [hunk 1]
```diff
@@ -61,14 +61,28 @@ def get_unique_affiliations(authors):
...
+    extracts the file extension and raises a ValueError if the extension is not one of our supported formats.
+    """
+    extension = os.path.splitext(filename)
+    if extension in supported_formats:
+        return extension
+    else:
+        raise ValueError(f'Extension of given output file ({filename}) is not one of the supported formats. Formats currently supported: ({supported_formats}).')
+
+
+def main_procedure(cff_path, init_path):
```
Comment (tobyhodges) [inline src/cff2pages/cff2pages.py:75]: Note that I am switching the order of the parameters here

**Comment by B3J4y** [comment]
I don't know if you have gotten a message, @tobyhodges. Can you handle the failing checks?

**Comment by tobyhodges** [comment]
@B3J4y I have updated and expanded the tests. Two are failing, but that seems to be due to a mismatch in the expected and actual HTML outputs (`CurrentCffTester.test_generated_current_html_body` and `MinimalCffTester.test_generated_minimal_html_body`): not sure these failures are related to the changes here?

A reminder that this PR should be squash-merged: I made a real mess of the commit history along the way – sorry!

File: tests/test_cff2pages.py [hunk 2]
```diff
@@ -62,13 +63,13 @@ def test_authors_affiliation(self):
...
     <div class="content">
-        
+
         <h1 class="blog-title"> Test CFF</h1>
 
-        
+
     <h2>
             Muster Mina
-            <sup>2</sup>,
```
Comment (tobyhodges) [inline tests/test_cff2pages.py:71]: Are the test failures related to my text editor stripping out extraneous whitespace, @B3J4y?

**Pull request merged**